modules/collect/src/main/java/com/opengamma/collect/result/Result.java

```diff
@@ -304,6 +306,39 @@
     return new Result<>(Failure.of(items));
   }
 
+  /**
+   * Returns a success result containing the value if it is non-null, else returns a failure result
+   * with the specified reason and message.
+   * <p>
+   * This is useful for interoperability with APIs that return {@code null}, for example {@code Map.get()}, where
+   * a missing value should be treated as a failure.
+   * <p>
+   * The message is produced using a template that contains zero to many "{}" placeholders.
+   * Each placeholder is replaced by the next available argument.
+   * If there are too few arguments, then the message will be left with placeholders.
+   * If there are too many arguments, then the excess arguments are appended to the
+   * end of the message. No attempt is made to format the arguments.
+   * See {@link Messages#format(String, Object...)} for more details.
+   *
+   * @param <R> the expected type of the result
+   * @param reason  the reason for the failure
+   * @param message  a message explaining the failure, uses "{}" for inserting {@code messageArgs}
+   * @param messageArgs  the arguments for the message
+   * @return a success result if the value is non-null, else a failure result
+   */
+  public static <R> Result<R> ofNullable(
```

Comment:
Is it worth having an alternative where the reason and message are defaulted - if you're not expecting failures then passing 3 args each time is quite painful (and the message is likely to be "this can't happen!")